Arm/BitVec.lean

```diff
@@ -636,12 +629,11 @@ theorem append_of_extract_general_nat (high low n vn : Nat) (h : vn < 2 ^ n) :
   done
 
 theorem append_of_extract (n : Nat) (v : BitVec n)
-  (high0 : high = n - low) (low0 : 1 <= low)
-  (h : high + (low - 1 - 0 + 1) = n) :
-  BitVec.cast h (zeroExtend high (v >>> low) ++ extractLsb (low - 1) 0 v) = v := by
+  (high0 : high = n - low) (h : high + low = n) :
+  BitVec.cast h (zeroExtend high (v >>> low) ++ extractLsb' 0 low v) = v := by
```

Comment:
This reminds me: this'd probably be a better rule if we put the cast on the RHS. Mind doing that please, @pennyannn? :)

Reply:
Hi @shigoel , I tried this and it broke the proof for `read_mem_bytes_of_write_mem_bytes_same` which has a goal that looks like:
```
n✝ n : Nat
     hn : Nat.succ 0 ≤ n
     n_ih : ∀ {addr : BitVec 64} {v : BitVec (n * 8)} {s : ArmState},
       n ≤ 2 ^ 64 → ¬n = 0 → read_mem_bytes n addr (write_mem_bytes n addr v s) = v
     addr : BitVec 64
     v : BitVec ((n + 1) * 8)
     s : ArmState
     hn1 : n + 1 ≤ 2 ^ 64
     hn0 : ¬n + 1 = 0
     ⊢ BitVec.cast ⋯ (zeroExtend (n * 8) (v >>> 8) ++ read_mem addr (write_mem addr (extractLsb' 0 8 v) s)) = v
```
Notice how the `BitVec.cast` is on the LHS. 
Could you shed some light for why putting `BitVec.cast` on RHS would be better?

Reply:
Ah, sorry! I was going by the rule of thumb that the LHS should be general enough to match and maybe the cast was limiting it. In this case, it turns out this rule was aiding a very specific theorem, so maybe the change I suggested was not great.

Sorry for the noise then!

Reply:
It seems like it is only used that theorem in memory proofs. Maybe let's see if there will be more use later.